### PR TITLE
Generate yarn exclude file using template

### PR DESCRIPTION
**Pull request by amithkanand**
This PR generates `yarn.exclude` file using correct template.

### CODE REVIEW

**Comment by cbaenziger** [comment]
I would like to see this stay a `file` resource since the format is very simple. Further, we should not be editing templates for cluster operations. Templates are code and with the merge of the friendly branch we have moved away from editing code for operational changes.

**Comment by amithkanand** [comment]
Changed it back to a `file` resource and added a new attribute for decommissioning hosts.

File: cookbooks/bcpc-hadoop/recipes/yarn_config.rb [hunk 1]
```diff
@@ -3,6 +3,9 @@
 include_recipe 'bcpc-hadoop::yarn_site'
 
 file "/etc/hadoop/conf/yarn.exclude" do
+  content node["bcpc"]["hadoop"]["decommission"]["hosts"].each {|hst| puts hst}
```
Comment (kiiranh) [inline cookbooks/bcpc-hadoop/recipes/yarn_config.rb]: Nit: For readability, `|host| puts host` is much better than `|hst| puts hst`.

**Comment by amithkanand** [comment]
This is tested now and is working.

**Comment by bijugs** [comment]
@amithkanand , this change tested in a new VM cluster? It fails since user ``yarn`` is not on the nodes when this resource executes. User ``yarn`` I believe gets created when ``yarn`` software gets installed. Refer issue #615 for details.

**Pull request merged**